cime_config/machines/config_machines.xml

```diff
@@ -3613,11 +3613,8 @@
     <NTEST_PARALLEL_JOBS>4</NTEST_PARALLEL_JOBS>
     <BATCH_SYSTEM>lsf</BATCH_SYSTEM>
     <SUPPORTED_BY>e3sm</SUPPORTED_BY>
-    <MAX_TASKS_PER_NODE>84</MAX_TASKS_PER_NODE>
-    <MAX_MPITASKS_PER_NODE>84</MAX_MPITASKS_PER_NODE>
-    <MAX_MPITASKS_PER_NODE compiler="pgigpu">18</MAX_MPITASKS_PER_NODE>
-    <MAX_MPITASKS_PER_NODE compiler="gnugpu">42</MAX_MPITASKS_PER_NODE>
-    <MAX_MPITASKS_PER_NODE compiler="ibmgpu">42</MAX_MPITASKS_PER_NODE>
+    <MAX_TASKS_PER_NODE>42</MAX_TASKS_PER_NODE>
```

Comment:
Now I'm really confused. If there are 42 cores on each summit node, wouldn't the old default of 84 MPIs/node cause crashes? And doesn't MAX_TASKS_PER_NODE need to equal the number of GPUs per node for GPU runs? I'm confused how it was set to 42 instead of 6 in the past...

Reply:
Sorry for the confusion, these lines are not needed here in summit-cpu machine since it doesn't have these *gpu compilers, I should remove these lines before. For cpu runs on Summit, we need to use full cpu cores (42), but for GPU runs, we only use 6-mpirank with one mpirank per GPU, together with the CPU OpenMP for full useage of Summit CPU.

Reply:
The IBM Power 9's have support for hardware threads in different models. There are 42 physical cores which can support 1,2 and 4 h/w threads each in the SMT1, SMT2 and SMT4 mode respectively. So 84 MPI tasks per node ideally should use SMT2 mode. Long ago in regular E3SM testing, I found SMT2 mode to be the best fit for E3SM workloads. 

For GPU workloads, we can't use more MPI ranks than GPUs, so we need to limit to 6. Hope this helps

Reply:
This max_tasks_* shouldn't be in effect if we configured the job to use _-pecount option_

Reply:
Ok, that all makes perfect sense. Thanks! So should we change to SMT2 in this PR?